Pcap++/src/PcapFilter.cpp

```diff
@@ -9,10 +9,62 @@
 #elif LINUX
 #include <in.h>
 #endif
+#include <pcap.h>
+#include "RawPacket.h"
 
 namespace pcpp
 {
 
+FilterTester::FilterTester(const std::string &filterStr) : filterStr(filterStr), prog(nullptr)

```

Comment:
setting a smart pointer with `nullptr` doesn't work in majority of the platform, please see the CI builds that fail on most platforms

Reply:
DIdn't know this, I can remove the smart pointer and use old school ptr here and just de-allocate it in destructor. 

Is there a cross platform compatible way to handle this?

Reply:
I'm not sure if there's a cross platform way to do it with smart pointers. But if you're changing it anyway to standard pointers that shouldn't be an issue anymore